WorkloadManagementSystem/PilotAgent/pilotCommands.py

```diff
@@ -359,6 +358,94 @@ def _getSecurityCFG( self ):
       self.cfg.append( "-o /DIRAC/Security/CertFile=%s/hostcert.pem" % self.pp.certsLocation )
       self.cfg.append( "-o /DIRAC/Security/KeyFile=%s/hostkey.pem" % self.pp.certsLocation )
 
+class CheckCECapabilities( CommandBase ):
+  """ Used to get  CE tags.
+  """
+  def __init__( self, pilotParams ):
+    """ c'tor
+    """
+    super( CheckCECapabilities, self ).__init__( pilotParams )
+
+    # this variable contains the options that are passed to dirac-configure, and that will fill the local dirac.cfg file
+    self.cfg = []
+
+  def execute( self ):
+    """ Setup CE/Queue Tags
+    """
+
+    if self.pp.useServerCertificate:
+      self.cfg.append( '-o  /DIRAC/Security/UseServerCertificate=yes' )
+    if self.pp.localConfigFile:
+      self.cfg.append( self.pp.localConfigFile )  # this file is as input
+
+
+    checkCmd = 'dirac-resource-get-parameters -S %s -N %s -Q %s %s' % ( self.pp.site, self.pp.ceName, self.pp.queueName,
+                                                                               " ".join( self.cfg ) )
+    retCode, resourceDict = self.executeAndGetOutput( checkCmd, self.pp.installEnv )
+
+    import json
+    resourceDict = json.loads( resourceDict )
+    if retCode:
+      self.log.warn( "Could not get resource parameters [ERROR %d]" % retCode )
+    if resourceDict.get( 'Tag' ):
+      self.pp.tags.append( resourceDict['Tag'] )
+      self.cfg.append( '-FDMH' )
+      if self.pp.localConfigFile:
+        self.cfg.append( '-O %s' % self.pp.localConfigFile )
+        self.cfg.append( self.pp.localConfigFile )
+
+      if self.debugFlag:
+        self.cfg.append( '-ddd' )
+
+      self.cfg.append( '-o "/Resources/Computing/CEDefaults/Tag=%s"' % ','.join( self.pp.tags ) )
+
+      configureCmd = "%s %s" % ( self.pp.configureScript, " ".join( self.cfg ) )
+      retCode, _configureOutData = self.executeAndGetOutput( configureCmd, self.pp.installEnv )
+      if retCode:
+        self.log.warn( "Could not configure DIRAC [ERROR %d]" % retCode )
+
+class CheckWNCapabilities( CommandBase ):
+  """ Used to get  CE tags.
+  """
+  def __init__( self, pilotParams ):
+    """ c'tor
+    """
+    super( CheckWNCapabilities, self ).__init__( pilotParams )
+    self.cfg = []
+
+  def execute( self ):
+    """ Discover #Processors and memory
+    """
+
+
+    retCode, result = self.executeAndGetOutput( 'dirac-wms-get-wn-parameters' , self.pp.installEnv )
+    result = result.split( ' ' )
+    Processors = result[0]
+    totalMemory = result[1]
+    if not Processors:
+      self.log.warn( "Could not retrieve number of processors" )
+    else:
+      self.pp.tags.append( 'Processors_%s' % Processors )
```

Comment:
Hi Andrei, I will modify it according to your comment. Following this JIRA task https://its.cern.ch/jira/browse/LHCBDIRAC-468 I was using tags. I would really appreciate to discuss better the tasks to be sure we agree on what it is needed to be done.